lib/sources/eth-erc20/index.ts

```diff
@@ -0,0 +1,224 @@
+import 'dotenv/config';
+import { Logger } from 'pino';
+import pLimit from 'p-limit';
+import { CbOnUserBalances } from '../../../types/sources/cbOnUserBalances';
+import { SourceInterface } from '../../../types/sources/source';
+import { Database } from 'bun:sqlite';
+
+const viem = require('viem');
+
+const ERC20_ABI = [
+  {
+    inputs: [{ internalType: 'address', name: 'owner', type: 'address' }],
+    name: 'balanceOf',
+    outputs: [{ internalType: 'uint256', name: '', type: 'uint256' }],
+    stateMutability: 'view',
+    type: 'function',
+  },
+];
+
+export default class EthERC20Source<A = { contract: string }>
+  implements SourceInterface
+{
+  rpc: string;
+  concurrencyLimit: number;
+  paginationLimit: number;
+  logger: Logger<never>;
+  assets: Record<string, { contract: string } & A>;
+  sourceName: string;
+  client: any;
+  db?: Database;
+  networkName: string;
+
+  getClient = () => {
+    if (!this.client) {
+      this.client = viem.createPublicClient({
+        transport: viem.http(this.rpc),
+      });
+    }
+    return this.client;
+  };
+
+  constructor(rpc: string, logger: Logger<never>, params: any, db?: Database) {
```

Comment:
https://github.com/hadronlabs-org/drop-points-crawler/blob/2665decd4e294fd04c9a6ad39c4483480e6e2783/crawler/commands/crawl.ts#L100C29-L102

pretty sure it won't work